### DISCUSSION TITLE
Copy / Paste does not work

### DISCUSSION THREAD

**Discussion by halkingb** [aclap-dev/video-downloadhelper#1498] [Help and Q&A]
### Your question

Whenever I download a video and the dialog opens where I choose the location and name of the file, I cannot (!) copy or paste file names from / to the dialog window. I need to type everything by hand.

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: de
coapp: {"found":true,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: aarch64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:132.0) Gecko/20100101 Firefox/132.0
{
    "used_history_button": true,
    "download_directory": "/Users/halkingb/Downloads",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "/Users/halkingb/Downloads"
        }
    },
    "concurrent_downloads_max": 40,
    "show_thumbnail_in_notification": false,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": false,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "default_action": "download_as",
    "yt_warning": false,
    "open_count_store": 77,
    "successfull_dl": 34,
    "never_show_successfull_dl_message": true
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

That has to be a Mac problem.  I paste things from the clipboard into the file selection dialog on Windows all the time & it works just fine.  You might want to investigate some Mac forums & look for other reports of a similar problem in contexts entirely unrelated to VDH.

**Comment by Mattmjs** [comment]
@halkingb you provided all those details but deleted one crucial detail. Here is what mine says.

 "http_media_download_strategy": "inbrowser",

That's normal behavior you are seeing if you have this setting. In that case copy the file name to the clipboard first before you
open the save as dialog.

If it's the companion app then you can copy and paste file names while the dialog is open like GrampaWildWilly says.

**Reply by halkingb** [reply]
Yeah, unfortunately it doesn't matter whether it's the companion app or not (tried both, doesn't work either way).

Even if I pre-save something to the clipboard before opening the dialog, it doesn't let me paste anything.

I even went back to the **legacy version** to check the settings. Everything works as it should there.
